sql: mismatch between MySQL version and CRDB in pg_metadata_test

**Describe the problem**

In https://github.com/cockroachdb/cockroach/pull/84776, I added a new column`is_visible` to the table `information_schema.statistics`. 
[GitHub CI](https://teamcity.cockroachdb.com/viewLog.html?buildId=5825397&buildTypeId=Cockroach_UnitTests) failed due to `TestInformationSchemaMySQL` under `pkg/sql/pg_metadata_test.go`.

I spoke with @RichardJCai  about this, and he mentioned these two issues.
1. `pg_metadata_test.go` should give a better error message.
Running the suggested command `go test -run TestDiffTool --rewrite-diffs` or 
`go test -run TestDiffTool --add-missing-tables` was not helpful because the option
only works for tables from Postgres. 
2. There is a mismatch between the CRDB and MySQL version now.

**To Reproduce**

- Check out the branch from
https://github.com/wenyihu6/cockroach/tree/1.5-add-visible-schema 
- Run  `cd pkg/sql` 
- Run `go test -run TestInformationSchemaMySQL -show-logs`

**Expected behavior**

The GitHub CI should not fail, and a better error message should be given.

**Additional data / screenshots** 
- `is_visible` indicates if an index is visible to the optimizer. This feature was introduced to [MySQL 8.0](https://dev.mysql.com/doc/mysql-infoschema-excerpt/8.0/en/information-schema-statistics-table.html). This feature is not available in PostgreSQL, so information schema in PostgreSQL does not include this column. 
- I tried running this failed test case with `-show-logs`, and this was the output.
<details>

```
wenyihu@crlMBP-C02ZV00XMD6TMjA5 sql % go test -run TestInformationSchemaMySQL -show-logs
I220722 16:44:47.519393 1 (gostd) rand.go:176  [-] 1  random seed: 7625152197961646903
I220722 16:44:47.524065 10 testutils/serverutils/test_server_shim.go:115  [-] 1  Running test with the default test tenant. If you are only seeing a test case failure when this message appears, there may be a problem with your test case running within tenants.
I220722 16:44:47.525674 10 server/config.go:685  [n?] 2  1 storage engine initialized
I220722 16:44:47.525737 10 server/config.go:688  [n?] 3  Pebble cache size: 128 MiB
I220722 16:44:47.525751 10 server/config.go:688  [n?] 4  store 0: in-memory, size 512 MiB
I220722 16:44:47.527574 10 1@rpc/tls.go:268  [n?] 5  server certificate addresses: IP=127.0.0.1,::1; DNS=localhost,*.local; CN=node
I220722 16:44:47.527601 10 1@rpc/tls.go:317  [n?] 6  web UI certificate addresses: IP=127.0.0.1,::1; DNS=localhost,*.local; CN=node
W220722 16:44:47.527760 10 server/status/runtime.go:325  [n?] 7  could not parse build timestamp: parsing time "" as "2006/01/02 15:04:05": cannot parse "" as "2006"
I220722 16:44:47.560442 10 1@server/clock_monotonicity.go:65  [n?] 8  monitoring forward clock jumps based on server.clock.forward_jump_check_enabled
I220722 16:44:47.567755 10 server/init.go:197  [n?] 9  no stores initialized
I220722 16:44:47.567803 10 server/init.go:198  [n?] 10  awaiting `cockroach init` or join with an already initialized node
I220722 16:44:47.567820 10 server/init.go:262  [n?] 11  cluster 05cfe053-4a5b-4297-95a6-0e87cd27d2ff has been created
I220722 16:44:47.567843 10 server/init.go:263  [n?] 12  allocated node ID: n1 (for self)
I220722 16:44:47.567856 10 server/init.go:264  [n?] 13  active cluster version: 22.1-28
I220722 16:44:47.567978 198 1@server/server.go:1326  [n1] 14  connecting to gossip network to verify cluster ID "05cfe053-4a5b-4297-95a6-0e87cd27d2ff"
W220722 16:44:47.567992 10 1@gossip/gossip.go:1465  [n1] 15  no addresses found; use --join to specify a connected node
I220722 16:44:47.568120 10 gossip/gossip.go:403  [n1] 16  NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:54009" > attrs:<> locality:<> ServerVersion:<major_val:22 minor_val:1 patch:0 internal:28 > build_tag:"unknown" started_at:1658508287568111000 cluster_name:"" sql_address:<network_field:"tcp" address_field:"127.0.0.1:54010" > http_address:<network_field:"tcp" address_field:"127.0.0.1:54008" >
I220722 16:44:47.571939 10 server/node.go:466  [n1] 17  initialized store s1
I220722 16:44:47.572054 10 kv/kvserver/stores.go:279  [n1] 18  read 0 node addresses from persistent storage
I220722 16:44:47.572107 198 1@server/server.go:1329  [n1] 19  node connected via gossip
I220722 16:44:47.572160 10 server/node.go:545  [n1] 20  started with engine type 2
I220722 16:44:47.572191 10 server/node.go:547  [n1] 21  started with attributes []
I220722 16:44:47.572332 10 1@server/server.go:1465  [n1] 22  starting https server at 127.0.0.1:54008 (use: 127.0.0.1:54008)
I220722 16:44:47.572368 10 1@server/server.go:1470  [n1] 23  starting postgres server at 127.0.0.1:54010 (use: 127.0.0.1:54010)
I220722 16:44:47.572397 10 1@server/server.go:1472  [n1] 24  starting grpc server at 127.0.0.1:54009
I220722 16:44:47.572428 10 1@server/server.go:1473  [n1] 25  advertising CockroachDB node at 127.0.0.1:54009
W220722 16:44:47.604864 38 kv/range_lookup.go:243  [n1,rangefeed=table-stats-cache,range-lookup=/Meta2/Table/20/NULL] 26  range lookup of key /Meta2/Table/20/NULL found only non-matching ranges []; retrying
W220722 16:44:47.606294 37 kv/range_lookup.go:243  [n1,rangefeed=table-stats-cache,range-lookup=/Table/20] 27  range lookup of key /Table/20 found only non-matching ranges []; retrying
I220722 16:44:47.660336 423 kv/kvserver/replica_rangefeed.go:689  [n1,rangefeed=table-stats-cache,s1,r21/1:/Table/2{0-1}] 28  RangeFeed closed timestamp is empty
I220722 16:44:47.719478 60 kv/kvclient/rangefeed/rangefeedcache/watcher.go:316  [n1] 29  spanconfig-subscriber: established range feed cache
I220722 16:44:47.719826 10 1@util/log/event_log.go:32  [n1] 30 ={"Timestamp":1658508287719823000,"EventType":"node_join","NodeID":1,"StartedAt":1658508287568111000,"LastUp":1658508287568111000}
I220722 16:44:47.719866 10 sql/sqlliveness/slinstance/slinstance.go:327  [n1] 31  starting SQL liveness instance
I220722 16:44:47.720012 450 sql/temporary_schema.go:515  [n1] 32  running temporary object cleanup background job
I220722 16:44:47.720097 452 sql/sqlstats/persistedsqlstats/provider.go:129  [n1] 33  starting sql-stats-worker with initial delay: 9m5.421915961s
I220722 16:44:47.721250 64 sql/sqlliveness/slstorage/slstorage.go:406  [n1] 34  inserted sqlliveness session 5acbc1ae82424c1daf58bd9c0e7f4049
I220722 16:44:47.728382 64 sql/sqlliveness/slinstance/slinstance.go:205  [n1] 37  created new SQL liveness session 5acbc1ae82424c1daf58bd9c0e7f4049
I220722 16:44:47.721658 450 sql/temporary_schema.go:560  [n1] 35  found 0 temporary schemas
I220722 16:44:47.727815 659 kv/kvclient/rangefeed/rangefeedcache/watcher.go:316  [n1] 36  settings-watcher: established range feed cache
I220722 16:44:47.728442 450 sql/temporary_schema.go:563  [n1] 38  early exiting temporary schema cleaner as no temporary schemas were found
I220722 16:44:47.728487 450 sql/temporary_schema.go:564  [n1] 39  completed temporary object cleanup job
I220722 16:44:47.728597 223 kv/kvclient/rangefeed/rangefeedcache/watcher.go:316  [n1] 40  system-config-cache: established range feed cache
I220722 16:44:47.729147 450 sql/temporary_schema.go:649  [n1] 41  temporary object cleaner next scheduled to run at 2022-07-22 17:14:47.719958 +0000 UTC m=+1800.321080747
I220722 16:44:47.732621 658 spanconfig/spanconfigmanager/manager.go:147  [n1] 42  started auto span config reconciliation job
I220722 16:44:47.732690 653 jobs/adopt.go:242  [n1] 43  job 781313270506422273: resuming execution
I220722 16:44:47.737294 656 jobs/registry.go:1208  [n1] 44  AUTO SPAN CONFIG RECONCILIATION job 781313270506422273: stepping through state running with error: <nil>
I220722 16:44:47.739710 656 jobs/registry.go:1425  [n1] 45  AUTO SPAN CONFIG RECONCILIATION job 781313270506422273: toggling idleness to true
I220722 16:44:47.768147 656 spanconfig/spanconfigsqlwatcher/sqlwatcher.go:263  [n1,job=781313270506422273] 46  established range feed over system.descriptors starting at time 1658508287.745772000,0
I220722 16:44:47.768294 656 spanconfig/spanconfigsqlwatcher/sqlwatcher.go:321  [n1,job=781313270506422273] 47  established range feed over system.zones starting at time 1658508287.745772000,0
I220722 16:44:47.768364 656 spanconfig/spanconfigsqlwatcher/sqlwatcher.go:416  [n1,job=781313270506422273] 48  established range feed over system.protected_ts_records starting at time 1658508287.745772000,0
I220722 16:44:47.802697 620 util/log/event_log.go:32  [n1,intExec=optInToDiagnosticsStatReporting] 49 ={"Timestamp":1658508287729767000,"EventType":"set_cluster_setting","Statement":"SET CLUSTER SETTING \"diagnostics.reporting.enabled\" = true","Tag":"SET CLUSTER SETTING","User":"root","ApplicationName":"$ internal-optInToDiagnosticsStatReporting","SettingName":"diagnostics.reporting.enabled","Value":"true"}
I220722 16:44:47.805454 696 upgrade/upgrademanager/manager.go:110  [n1,intExec=set-setting,migration-mgr] 50  no need to migrate, cluster already at newest version
I220722 16:44:47.806534 696 util/log/event_log.go:32  [n1,intExec=set-setting] 51 ={"Timestamp":1658508287804573000,"EventType":"set_cluster_setting","Statement":"SET CLUSTER SETTING version = $1","Tag":"SET CLUSTER SETTING","User":"root","ApplicationName":"$ internal-set-setting","PlaceholderValues":["'22.1-28'"],"SettingName":"version","Value":"22.1-28"}
I220722 16:44:47.817732 844 util/log/event_log.go:32  [n1,intExec=initializeClusterSecret] 52 ={"Timestamp":1658508287815315000,"EventType":"set_cluster_setting","Statement":"SET CLUSTER SETTING \"cluster.secret\" = gen_random_uuid()::STRING","Tag":"SET CLUSTER SETTING","User":"root","ApplicationName":"$ internal-initializeClusterSecret","SettingName":"cluster.secret","Value":"ec6b10bf-dc2f-4560-b7c7-bf9943120613"}
I220722 16:44:47.821474 878 5@util/log/event_log.go:32  [n1,intExec=create-default-DB] 53 ={"Timestamp":1658508287818732000,"EventType":"create_database","Statement":"CREATE DATABASE IF NOT EXISTS defaultdb","Tag":"CREATE DATABASE","User":"root","DescriptorID":100,"ApplicationName":"$ internal-create-default-DB","DatabaseName":"defaultdb"}
I220722 16:44:47.824114 898 5@util/log/event_log.go:32  [n1,intExec=create-default-DB] 54 ={"Timestamp":1658508287821655000,"EventType":"create_database","Statement":"CREATE DATABASE IF NOT EXISTS postgres","Tag":"CREATE DATABASE","User":"root","DescriptorID":102,"ApplicationName":"$ internal-create-default-DB","DatabaseName":"postgres"}
I220722 16:44:47.835238 10 server/server_sql.go:1322  [n1] 55  done ensuring all necessary startup migrations have run
I220722 16:44:47.835351 864 jobs/job_scheduler.go:433  [n1] 56  waiting 3m0s before scheduled jobs daemon start
I220722 16:44:47.835434 964 kv/kvclient/rangefeed/rangefeedcache/watcher.go:316  [n1] 57  tenant-settings-watcher: established range feed cache
I220722 16:44:47.837250 963 server/auto_upgrade.go:59  [n1] 58  no need to upgrade, cluster already at the newest version
I220722 16:44:47.847290 10 1@server/server_sql.go:1392  [n1] 59  serving sql connections
W220722 16:44:47.878345 64 sql/sqlliveness/slinstance/slinstance.go:242  [n1] 60  exiting heartbeat loop
I220722 16:44:47.878359 522 sql/stats/automatic_stats.go:504  [n1] 61  quiescing auto stats refresher
W220722 16:44:47.878427 857 kv/kvserver/intentresolver/intent_resolver.go:796  [-] 62  failed to gc transaction record: could not GC completed transaction anchored at /NamespaceTable/30/1/100/0/"public"/4/1: node unavailable; try another peer
W220722 16:44:47.878504 527 jobs/registry.go:817  [n1] 63  canceling all adopted jobs due to stopper quiescing
I220722 16:44:47.878618 656 jobs/registry.go:1208  [n1] 64  AUTO SPAN CONFIG RECONCILIATION job 781313270506422273: stepping through state succeeded with error: <nil>
W220722 16:44:47.878669 656 kv/txn.go:701  [n1] 65  failure aborting transaction: node unavailable; try another peer; abort caused by: context canceled
test logs left over in: /var/folders/lq/w8d89b6d4vl5q1wblwhwv7rw0000gq/T/logTestInformationSchemaMySQL1089484724
--- FAIL: TestInformationSchemaMySQL (0.41s)
    test_log_scope.go:162: test logs captured to: /var/folders/lq/w8d89b6d4vl5q1wblwhwv7rw0000gq/T/logTestInformationSchemaMySQL1089484724
    --- FAIL: TestInformationSchemaMySQL/Table=statistics (0.00s)
        pg_metadata_test.go:356: Diff EXPECTED. To fix this, please run: cd pkg/sql; go test -run TestDiffTool --rewrite-diffs
    pg_metadata_test.go:1538: -- test log scope end --
FAIL
exit status 1
FAIL    github.com/cockroachdb/cockroach/pkg/sql        2.338s
```

</details>

- Here is a screenshot for the GitHub CI failure. 
<img width="1570" alt="Screen Shot 2022-07-22 at 1 52 23 PM" src="https://user-images.githubusercontent.com/56973754/180496209-8bb6f7da-787b-4ff7-b6b0-be325e6726dc.png">

Jira issue: CRDB-17947

--- a/pkg/sql/pg_metadata_test.go
+++ b/pkg/sql/pg_metadata_test.go
@@ -130,7 +130,6 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/sql/sem/tree"
 	"github.com/cockroachdb/cockroach/pkg/sql/types"
 	"github.com/cockroachdb/cockroach/pkg/testutils/serverutils"
-	"github.com/cockroachdb/cockroach/pkg/testutils/skip"
 	"github.com/cockroachdb/cockroach/pkg/testutils/sqlutils"
 	"github.com/cockroachdb/cockroach/pkg/util/leaktest"
 	"github.com/cockroachdb/cockroach/pkg/util/log"
@@ -1543,7 +1542,6 @@ func TestInformationSchemaPostgres(t *testing.T) {
 // NOTE: --catalog or --rdbms flags won't take effect on this test.
 func TestInformationSchemaMySQL(t *testing.T) {
 	defer leaktest.AfterTest(t)()
-	skip.WithIssue(t, 84915, "mismatch between MySQL version and CRDB")
 	defer log.Scope(t).Close(t)
 
 	NewDiffTool(t).Catalog("information_schema").RDBMS(MySQL).Run()

--- a/pkg/sql/testdata/mysql_test_expected_diffs_on_information_schema.json
+++ b/pkg/sql/testdata/mysql_test_expected_diffs_on_information_schema.json
@@ -1,10 +1,10 @@
 {
-  "version": "8.0.25",
+  "version": "8.0.27",
   "diffSummary": {
     "TotalTables": 48,
     "TotalColumns": 476,
     "MissingTables": 0,
-    "MissingColumns": 80,
+    "MissingColumns": 79,
     "DatatypeMismatches": 0
   },
   "diffs": {
@@ -85,7 +85,6 @@
       "expression": null,
       "index_comment": null,
       "index_type": null,
-      "is_visible": null,
       "nullable": null,
       "packed": null,
       "sub_part": null

